internal/streamingnode/server/walmanager/wal_lifetime.go

```diff
@@ -0,0 +1,161 @@
+package walmanager
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+
+	"github.com/milvus-io/milvus/internal/streamingnode/server/wal"
+	"github.com/milvus-io/milvus/internal/util/streamingutil/status"
+	"github.com/milvus-io/milvus/pkg/log"
+	"github.com/milvus-io/milvus/pkg/streaming/util/types"
+)
+
+// newWALLifetime create a WALLifetime with opener.
+func newWALLifetime(opener wal.Opener, channel string) *walLifetime {
+	ctx, cancel := context.WithCancel(context.Background())
+	l := &walLifetime{
+		ctx:       ctx,
+		cancel:    cancel,
+		finish:    make(chan struct{}),
+		opener:    opener,
+		statePair: newWALStatePair(),
+		logger:    log.With(zap.String("channel", channel)),
+	}
+	go l.backgroundTask()
+	return l
+}
+
+// walLifetime is the lifetime management of a wal.
+// It promise a wal is keep state consistency in distributed environment.
+// All operation on wal management will be sorted with following rules:
+// (term, available) illuminate the state of wal.
+// term is always increasing, available is always before unavailable in same term, such as:
+// (-1, false) -> (0, true) -> (1, true) -> (2, true) -> (3, false) -> (7, true) -> ...
+type walLifetime struct {
```

Comment:
How do we use a walLease  instead of walLifetime to manage the life time of wal instance?